### ISSUE TITLE
Bug: Unsafe assignment of type any[] in TypeScript 5.3

### ISSUE THREAD

**Issue by Stadly** [typescript-eslint/typescript-eslint#8064]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

After upgrading from TypeScript 5.2 to 5.3, ESLint gives an error that I think is incorrect.

```json
// tsconfig.json
{
  "compilerOptions": {
    "exactOptionalPropertyTypes": true,
    "strict": true
  }
}
```

```ts
// tailwind.config.ts
import type { Config } from "tailwindcss";

export default {
  content: [],
  theme: {
    extend: {},
  },
  plugins: [require("@tailwindcss/forms")],
} satisfies Config;
```

`npx eslint .` gives the following error:
```
/home/projects/stackblitz-starters-zre7x7/tailwind.config.ts
  8:3  error  Unsafe assignment of type any[] to a variable of type (PluginCreator | { handler: PluginCreator; config?: Partial<Config>; } | { (options: any): { handler: PluginCreator; config?: Partial<...>; }; __isOptionsFunction: true; } | undefined)[]  @typescript-eslint/no-unsafe-assignment
```

If you downgrade TypeScript from version 5.3.2 to 5.2.2 there is no error.

If you turn off `exactOptionalPropertyTypes` in `tsconfig.json` there is no error, which seems strange.

I've reported the issue to TypeScript, who say it is not a TypeScript error: https://github.com/microsoft/TypeScript/issues/56767

### Reproduction Repository Link

https://stackblitz.com/edit/stackblitz-starters-zre7x7?file=tailwind.config.ts

### Repro Steps

1) open the link
2) `npx eslint .`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.14.0` |
| `@typescript-eslint/parser`            | `6.14.0` |
| `@typescript-eslint/scope-manager`     | `6.14.0` |
| `@typescript-eslint/typescript-estree` | `6.14.0` |
| `@typescript-eslint/type-utils`        | `6.14.0` |
| `@typescript-eslint/utils`             | `6.14.0` |
| `TypeScript`                           | `5.3.3` |
| `ESLint`                               | `8.55.0` |
| `node`                                 | `18.18.0` |

**Comment by bradzacher** [comment]
`require` returns `any`. It always does.

So your `plugins` array is going to be inferred as `any[]`. So you're assigning an any array to a property that expects a non-any array. Which is an error in the rule.

The behaviour of the types that TS provides us may have changed between versions - but that doesn't change the fact that the error is correct here.

**Comment by Stadly** [comment]
I was able to fix it using `import` instead of `require`:

```ts
import type { Config } from "tailwindcss";
import * as forms from "@tailwindcss/forms";

export default {
  content: [],
  theme: {
    extend: {},
  },
  plugins: [forms],
} satisfies Config;
```
